rct229/rulesets/ashrae9012019/section21/section21rule8.py

```diff
@@ -0,0 +1,176 @@
+
+
+class Section21Rule8(RuleDefinitionListIndexedBase):
+    """Rule 8 of ASHRAE 90.1-2019 Appendix G Section 21 (Hot water loop)"""
+
+    def __init__(self):
+        super(Section21Rule8, self).__init__(
+            rmrs_used=produce_ruleset_model_instance(
+                USER=False, BASELINE_0=True, PROPOSED=False
+            ),
+            each_rule=Section21Rule8.HeatingFluidLoopRule(),
+            index_rmr=BASELINE_0,
+            id="21-8",
+            description="When the baseline building requires boilers, (for baseline system type = 1,5,7,11 and 12), HWST for the baseline building shall be reset using an outdoor air dry-bulb reset schedule. 180F at 20F OAT, 150Fat 50F OAT, ramped linerarly between 150F and 180F.",
+            ruleset_section_title="HVAC - Water Side",
+            standard_section="Section G3.1.3.3 Building System-Specific Modeling Requirements for the Baseline model",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_descriptions/0",
+            list_path="$.fluid_loops[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.BASELINE_0
+        hw_loop_zone_list_w_area_dict_b = get_hw_loop_zone_list_w_area(rmi_b)
+        boiler_loop_ids_b = find_all("boilers[*].loop", rmi_b)
+        boiler_fluid_loops_set_b = set(
+            boiler_loop_id
+            for boiler_loop_id in boiler_loop_ids_b
+            if boiler_loop_id in hw_loop_zone_list_w_area_dict_b
+        )
+        return len(boiler_fluid_loops_set_b) > 0
+
+    def create_data(self, context, data):
+        rmi_b = context.BASELINE_0
+        boiler_loop_ids = find_all("boilers[*].loop", rmi_b)
+        return {"loop_boiler_dict": boiler_loop_ids}
+
+    def list_filter(self, context_item, data):
+        fluid_loop_b = context_item.BASELINE_0
+        loop_boiler_dict = data["loop_boiler_dict"]
+        return fluid_loop_b["id"] in loop_boiler_dict
+
+    class HeatingFluidLoopRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section21Rule8.HeatingFluidLoopRule, self).__init__(
+                rmrs_used=produce_ruleset_model_instance(
+                    USER=False, BASELINE_0=True, PROPOSED=False
+                ),
+                required_fields={
+                    "$": ["heating_design_and_control"],
+                    "heating_design_and_control": [
+                        "temperature_reset_type",
+                        "outdoor_high_for_loop_supply_reset_temperature",
+                        "outdoor_low_for_loop_supply_reset_temperature",
+                        "loop_supply_temperature_at_outdoor_high",
+                        "loop_supply_temperature_at_outdoor_low",
+                    ],
+                },
+            )
+
+        def get_calc_vals(self, context, data=None):
+            fluid_loop_b = context.BASELINE_0
+            temperature_reset_type = fluid_loop_b["heating_design_and_control"][
```

Comment:
Could we add ```_b``` at the end of variable names in this method since these variables represent values from baseline?

Reply:
Good point. Addressed!